src/main/kotlin/camp/nextstep/edu/racingcar/Application.kt

```diff
@@ -1,27 +1,28 @@
 package camp.nextstep.edu.racingcar
 
-import camp.nextstep.edu.racingcar.interfaces.CommandLineInterface
+import camp.nextstep.edu.racingcar.interfaces.RacingSnapshot
+import camp.nextstep.edu.racingcar.interfaces.cli.RacingInputReader
+import camp.nextstep.edu.racingcar.interfaces.cli.RacingResultWriter
+import camp.nextstep.edu.racingcar.interfaces.cli.RacingSnapshotWriter
 import camp.nextstep.edu.racingcar.racing.Car
 import camp.nextstep.edu.racingcar.racing.Racing
 
 fun main() {
-    val cli = CommandLineInterface()
-
-    val carNames = cli.readCarNames()
-    val moveCount = cli.readMoveCount()
+    val (carNames, moveCount) = RacingInputReader.readRacingInputs()
 
     val participants = carNames.mapIndexed { i, name -> Car(i, name) }
-    val racing = Racing.new(participants, moveCount)
+    val racing = Racing(participants, moveCount)
 
     racing.start()
-    val carTracks = racing.result()
 
-    for (i in 0 until moveCount) {
-        for ((car, track) in carTracks) {
-            cli.draw(car, track, i)
-        }
-        cli.drawBlank()
+    for (movements in 1..moveCount) {
+        RacingSnapshotWriter.writeRacingSnapshot(
+            RacingSnapshot(
+                racing.carRacingEvents().mapValues { (_, events) -> events.take(movements) }
```

Comment:
take 함수 👍 

```suggestion
                racing.carRacingEvents(take = movements)
```
racing 객체에 메시지를 보내는 방법도 좋을것 같습니다.